src/components/WorkflowSelector.tsx

```diff
@@ -1,163 +1,146 @@
-                      {workflow.selectedApps.length > 3 
-                        ? `${workflow.selectedApps.length}` 
-                        : workflow.selectedApps.length}
-                    </span>
-                  </div>
-                </TooltipTrigger>
-                <TooltipContent className="max-w-[300px]">
-                  <p>
-                    {workflow.settings?.isAllowList 
-                      ? `Only allowing ${formatAppsList(workflow.selectedApps)}` 
-                      : `Blocking ${formatAppsList(workflow.selectedApps)}`}
-                  </p>
-                </TooltipContent>
-              </Tooltip>
-            </TooltipProvider>
-          )}
-        </div>
-      </div>
-    </Card>
+    <motion.div layout>
+      <ContextMenu modal={false}>
+        <ContextMenuTrigger>
+          <Badge
+            variant={isSelected ? 'default' : 'secondary'}
+            className={cn(
+              'cursor-pointer whitespace-nowrap overflow-hidden text-ellipsis px-4',
+              className
+            )}
+            onClick={onClick}
+            {...props}
+          >
+            {workflow.name}
+          </Badge>
+        </ContextMenuTrigger>
+        <ContextMenuContent>
+        <ContextMenuItem onClick={onSettingsClick} className="cursor-pointer">
+            <Settings className="mr-2 h-4 w-4" />
+            <span>Configure</span>
+          </ContextMenuItem>
+          <ContextMenuSeparator />
+          <ContextMenuItem onClick={onCreateClick} className="cursor-pointer">
+            <Plus className="mr-2 h-4 w-4" />
+            <span>New Preset</span>
+          </ContextMenuItem>
+        </ContextMenuContent>
+      </ContextMenu>
+    </motion.div>
   )
 }
 
 export function WorkflowSelector({ selectedId, onSelect }: WorkflowSelectorProps) {
-  const [isDialogOpen, setIsDialogOpen] = useState(false)
+  const scrollContainerRef = useRef<HTMLDivElement>(null)
   const [isEditDialogOpen, setIsEditDialogOpen] = useState(false)
   const [isCreateDialogOpen, setIsCreateDialogOpen] = useState(false)
   const [editingWorkflow, setEditingWorkflow] = useState<Workflow | null>(null)
   const [workflows, setWorkflows] = useState<Workflow[]>([])
+  const [showLeftMask, setShowLeftMask] = useState(false)
+
+  // Handle scroll position to determine if left mask should be shown
+  const handleScroll = () => {
+    if (scrollContainerRef.current) {
+      setShowLeftMask(scrollContainerRef.current.scrollLeft > 0)
+    }
+  }
 
-  // Load workflows from database
+  // Setup scroll event listener
+  useEffect(() => {
+    const container = scrollContainerRef.current
+    if (container) {
+      container.addEventListener('scroll', handleScroll)
+      return () => container.removeEventListener('scroll', handleScroll)
+    }
+  }, [])
+
+  // Auto-scroll selected workflow into view
+  useEffect(() => {
+    if (selectedId && scrollContainerRef.current) {
```

Comment:
nesting in code is hard to read and there's a few separate if statements nested here. 

I'd recommend using what are known as "guard" clauses here instead. 

So, 
```
if (selectedId && scrollContainerRef.current) {
```
Becomes
```suggestion
    if (!selectedId || !scrollContainerRef.current) return
```

The if statement "guards" against the condition where those variables do not exist and doesn't require nesting as the rest of the function stops executing when the "return" is called

Should be able to accomplish the same thing without nesting

Reply:
good feedback!